csaf_2.0/json_schema/csaf_json_schema.json

```diff
@@ -154,13 +154,127 @@
             "Cisco AnyConnect Secure Mobility Client 2.3.185"
           ]
         },
-        "cpe": {
-          "$comment": "TODO: This should have a full regular expression to enforce CPE syntax.",
-          "title": "Common Platform Enumeration representation",
-          "description": "The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.",
-          "type": "string",
-          "pattern": "^(cpe:2\\.3:[aho\\*\\-](:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\\*\\-]))(:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){4})|([c][pP][eE]:/[AHOaho]?(:[A-Za-z0-9\\._\\-~%]*){0,6})$",
-          "minLength": 5
+        "product_identification_helper": {
```

Comment:
I like the renaming @mprpic

Reply:
There is a big chance that `product_identifiers` are mistaken for `product_id` resp. the difference is not clear to the user. With the `_helper` it becomes more clear that it is a method the support the user in identifying the product.

Reply:
You both have a point. The dog sharing my flat reminds me to stop working now. I will suggest a third name tomorrow which we might all love. Helper I do not really like, and identifier seems to trump id (which is not what we intend).

Reply:
Perhaps each of the properties could simply be defined on `full_product_name_t` itself, or is there a particular reason to have them grouped into this container? `product_identification_helper` isn't listed among the required properties of `full_product_name_t` so I assume it would be fine to simply mark all of its properties as optional as well.